tofu/spectro/_rockingcurve.py

```diff
@@ -0,0 +1,345 @@
+    # psi_-H equivalent to (-ih, -ik, -il)
+    Fbmod = np.sqrt(
+        F_re**2 + F_im**2 - 2.*(F_im_cos*F_im_sin - F_re_cos*F_re_sin)
+    )    # fbmod
+
+    if Fmod == 0.:
+        Fmod == 1e-30
+    if Fbmod == 0.:
+        Fbmod == 1e-30
+
+    # ratio imaginary part and real part of the structure factor
+    kk = F_im/F_re
+    # rek = Real(kk)
+    rek = (F_re_cos*F_im_cos + F_re_sin*F_im_sin)/(F_re**2.)
+
+    # Re(psi) = psi' = -(4pi*e**2*F'_H)/(m*w**2*V) if 1/4piEps0 = 1
+    # Im(psi) = psi'' = -(4pi*e**2*F''_H)/(m*w**2*V)
+    # real part of psi_H
+    psi_re = (re*(lamb**2)*F_re)/(np.pi*V)    # psihp
+    # zero-order real part (averaged) TBF
+    psi0_dre = -re*(lamb**2)*(
+        6.*(Zo + dfo_re) + 3.*(Zsi + dfsi_re)
+        )/(np.pi*V)   # psiop
+    # zero-order imaginary part (averaged)
+    psi0_im = -re*(lamb**2)*(6.*fo_im + 3.*fsi_im)/(np.pi*V)    # psios
+
+    # Integrated reflectivity for crystals models: perfect (Darwin model) &
+    # ideally mosaic thick crystal
+    # -------------------------------------------------------------
+
+    R_per = Zo*F_re*re*lamb**2*(1. + abs(np.cos(2.*theta)))/(
+        6.*np.pi*V*np.sin(2.*theta)
+    )
+    R_mos = F_re**2*re**2*lamb**3*(1. + (np.cos(2.*theta))**2)/(
+        4.*mu*V**2*np.sin(2.*theta)
+    )
+
+    # Rocking curve and integrated reflectivity with "dynamical" model vs angle
+    # and for the 2 states of photon polarization
+    # -------------------------------------------------------------------------
+
+    # incident wave polarization (normal & parallel components)
+    polar = np.r_[1., abs(np.cos(2.*theta))]
+    # variables of simplification y, dy, g, L
+    g = psi0_im/(polar*psi_re)
+    y = np.linspace(-10., 10., 501)    # ay
+    dy = np.zeros(501) + 0.1    # day
+    al = np.full((2, 501), 0.)
+
+    power_ratio = np.full((al.shape), np.nan)    # phpo
+    th = np.full((al.shape), np.nan)    # phpo
+    rr = np.full((polar.shape), np.nan)
+    for i in range(al[:, 0].size):
+        al[i, ...] = (y**2 + g[i]**2 + np.sqrt(
+            (y**2 - g[i]**2 + kk**2 - 1.)**2 + 4.*(g[i]*y - rek)**2
+            ))/np.sqrt((kk**2 - 1.)**2 + 4.*rek**2)
+        # reflecting power P_DIN
+        power_ratio[i, ...] = (Fmod/Fbmod)*(
+            al[i, :] - np.sqrt((al[i, :]**2 - 1.))
+        )
+        power_ratiob = power_ratio[i, ...]
+        # intensity scale on the glancing angle (y=kk/g)
+        th[i, ...] = (y*polar[i]*psi_re - psi0_dre)/np.sin(2.*theta)
+        # integration of the power ratio over dy
+        rhy = np.sum(dy*power_ratiob)
+        # diffraction radiation
+        # r(i=0): normal component & r(i=1): parallel component
+        rr[i, ...] = (polar[i]*psi_re*rhy)/np.sin(2.*theta)
+    R_dyn = np.sum(rr)/2.
+    if R_dyn < 1e-7:
+        msg = (
+            "Please check the equations for integrated reflectivity:\n"
+            "the value of R_din ({}) is less than 1e-7.\n".format(R_dyn)
+        )
+        raise Exception(msg)
```

Comment:
Nice, to start making the routine modular, implement a sub-routine dedicate to computing the integrated reflectivity for all models, to have here something like:

```
    R_per = _comp_integrated_reflectivity(...., model='Darwin')
    R_mos = _comp_integrated_reflectivity(...., model='Mosaic')
    R_dyn = _comp_integrated_reflectivity(...., model='Dynamic')
```